nvflare/fuel/f3/drivers/driver.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (c) 2021-2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List
+
+from nvflare.fuel.f3.connection import Connection, ConnState
+from nvflare.fuel.f3.drivers.connector import Connector
+
+
+class ConnMonitor(ABC):
+    @abstractmethod
+    def state_change(self, connection: Connection):
+        """Driver state change notification, including new connections
+
+        Args:
+            connection: The connection that state has changed
+
+        Raises:
+            CommError: If any error happens while processing the frame
+        """
+        pass
+
+
+class Driver(ABC):
```

Comment:
should we move this into upper layer?
``nvflare/fuel/f3/driver.py``

Reply:
Driver is not directly used so not exposed.

Reply:
What is considered "directly used"?

If we want to enable different drivers for users in the next release, we move this file up?

Reply:
User shouldn't use Driver classes. The driver selection is through the URL like "grpc://localhost:8002".